java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Communicator.java

```diff
@@ -29,8 +30,54 @@ public final class Communicator implements AutoCloseable {
     private final Instance _instance;
 
     /**
-     * Destroy the communicator.
-     * This Java-only method overrides close in java.lang.AutoCloseable and does not throw any exception.
+     * Constructs a communicator with the specified options.
+     *
+     * @param initData the options for the new communicator
+     *
+     */
+    public Communicator(InitializationData initData) {
+        _instance = new Instance();
+        _instance.initialize(this, initData);
+
+        try {
+            _instance.finishSetup(this);
+        } catch (RuntimeException ex) {
+            _instance.destroy(false);
+            throw ex;
+        }
+    }
+
+    /**
+     * Constructs a communicator with default options.
+     *
+     */
```

Comment:
And here too, no need for trailing blank lines:
```suggestion
    /**
     * Constructs a communicator with default options.
     */
```